junit/src/main/java/org/kaazing/k3po/junit/rules/Latch.java

```diff
@@ -28,6 +29,8 @@
     private final CountDownLatch startable;
     private final CountDownLatch finished;
     private volatile Thread testThread;
+    
+    private final AtomicBoolean exceptionOccurred = new AtomicBoolean(false); 
```

Comment:
What are you trying to prevent, setting of multiple exceptions, or sending interrupt only once, or both.

** For the multiple exceptions **

I see three spots in the code setting the exception
1) sending notify barrier, (which only occurs if it can send it on the TCP connect, i.e. it is closed)
2) sending await barrier, (which only occurs if it can send it on the TCP connect, i.e. it is closed)
3) and in latch/code execution

I believe case 1 and case 2 can only happen if there is already an error or success case (sending a notify that isn't needed).  I would, therefore, suggest we change case 1 such that exceptions are ignored.  And change case 2, such that the exception merely bypasses the waiting on the latch (don't rethrow or save it)

Making this change is good and could be done regardless of adding the AtomicBoolean above.

** For multiple interrupts **

Even with the suggested changes above, it looks like the test thread might still be interrupted multiple times (because there is one place that calls `notifyException` and a different location that calls set exception, both of which then send the interrupt) ?  If this is an issue,  then please rename this variable to testThreadInterrupted or something similar.  I would also suggest refactoring the code to extract a shared method `interruptTestThread()` or something like that and have that be the sole place that messes with AtomicBoolean.  Both the methods that send the interrupt would call this new private method.